src/interval/interval.rs

```diff
@@ -159,6 +159,17 @@ impl Interval {
         })
     }
 
+    /// Creates an interval by inverting the given interval
+    /// e.g. Perfect fifth (C to G) becomes a perfect fourth (G to C)
+    pub fn invert(interval: &Self) -> Interval {
+        if interval.semitone_count == 12 {
+            Self::from_semitone(12).unwrap()
```

Comment:
suggestion: i think instead of unwrapping, returning an error from the `invert` function would be better, we should avoid unwrap whenever possible. That said, there might be a few unwrap's left in the library, they should also be cleaned up at some point.

Reply:
Good point, I'll make these functions more similar to `from_semitone` using the `IntervalError` struct.